### DISCUSSION TITLE
data/external/fitbit_csv directory missing

### DISCUSSION THREAD

**Discussion by sureshksampath** [carissalow/rapids#215] [Q&A]
Hello,

I am right now executing the RAPIDS configuration steps. I noticed that the directory 'data/external/fitbit_csv' is not available. Do I have to create it? 

Also, do I have to use the Fitbit Web API? Or, can I use the available test CSV files of Fitbit? If so, should I copy those files in   'data/external/fitbit_csv'?

Lastly, I assume I need to run the Container.R script available in '/rapids/src/data/streams/fitbitparsed_csv' after the CSV files are placed in the appropriate directory? Please clarify.

Thanks,
Suresh

**Comment by jenniferfedor** [comment]
Hi @sureshksampath,

Yes, if you would like to use the `fitbitjson_csv` or `fitbitparsed_csv` data streams to process your Fitbit data, you will need to specify the file path to these data. There are two options: 
- You can leave the relevant `FOLDER` parameter in your config file as-is, create the `fitbit_csv` directory in `data/external`, and copy your data into that directory
- You can change the relevant `FOLDER` parameter in your config file to point to the directory where your data are already stored 
 

Please feel free to refer to [these](https://www.rapids.science/1.9/datastreams/fitbitjson-csv/) [sections](https://www.rapids.science/1.9/datastreams/fitbitparsed-csv/) of our documentation for more information about those two data streams.  

<br>

If you would like to use our RAPIDS Fitbit test data to test execution of the pipeline, those data are already available in [`tests/data/external/aware_csv/fitbit_*.csv`](https://github.com/carissalow/rapids/tree/master/tests/data/external/aware_csv), so you will not need to use the Fitbit API to access them. You could copy those data into to the `data/external/fitbit_csv` directory you create or edit your config file to point to the location where those data are already stored. 

In addition, you will also likely want to do the following:
-	Change the Fitbit data stream to use in your config file to `fitbitparsed_csv`  
-	For each Fitbit sensor, change the `CONTAINER` parameter to the corresponding CSV file (e.g., `fitbit_calories_intraday_raw.csv` for the `FITBIT_CALORIES_INTRADAY` sensor)
-	Copy over the participant file for the test Fitbit device, [`tests/data/external/participant_files/fitbit.yaml`](https://github.com/carissalow/rapids/blob/master/tests/data/external/participant_files/fitbit.yaml), into your `data/external/participant_files` directory (you will also need to create this directory if you haven’t yet)
-	Update the PIDS section of your config file 

<br>

It should not be necessary to run the container.R script manually. As long as the correct data stream has been specified in the config file, RAPIDS will run the appropriate container.R script as part of the pipeline.  

<br>

I hope that helps! Please let us know if you have any additional questions. Thank you!
